src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -132,21 +146,32 @@ public class ProvenBlockHeaderStore : IProvenBlockHeaderStore
             this.asyncLoopFactory = asyncLoopFactory;
 
             this.performanceCounter = new BackendPerformanceCounter(dateTimeProvider);
-            nodeStats.RegisterStats(this.AddBenchStats, StatsType.Benchmark, 100);
+            nodeStats.RegisterStats(this.AddBenchStats, StatsType.Benchmark);
+            nodeStats.RegisterStats(this.AddComponentStats, StatsType.Component);
         }
 
         /// <inheritdoc />
-        public async Task InitializeAsync()
+        public async Task InitializeAsync(ChainedHeader chainedHeader)
         {
             await this.provenBlockHeaderRepository.InitializeAsync().ConfigureAwait(false);
 
-            this.TipHashHeight = this.provenBlockHeaderRepository.TipHashHeight;
+            this.storeTip = chainedHeader.Height == this.provenBlockHeaderRepository.TipHashHeight.Height ? chainedHeader : null;
+
+            if (this.storeTip == null)
+            {
+                this.storeTip = chainedHeader;
+
+                await this.RecoverStoreTipAsync(chainedHeader).ConfigureAwait(false);
```

Comment:
better make this method return new tip and then assign store tip to return value